libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,281 @@
+import { isNull, isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    ArrowRight as ArrowRightIcon,
+    Clear as ClearIcon,
+} from '@mui/icons-material/';
+import { Box, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useEffect, useRef, useState } from 'react';
+import { asFormattedMediaTimecodeString } from './shared/as-formatted-media-timecode-string';
+import { AudioMIMEType } from './shared/audio-mime-type.enum';
+import { isAudioMIMEType } from './shared/is-audio-mime-type';
+import { KeyboardKey, useKeyDown } from './shared/use-key-down';
+import {
+    TimeRangeSelectionStatus,
+    TimeRangeSelectionStatusIndicator,
+} from './time-range-selection-visual';
+
+type Nullable<T> = T | null;
+
+export type MedidaPlayDirection = 'forward' | 'reverse';
+
+export enum KeyboardShortcuts {
+    play = KeyboardKey.spacebar,
+    clear = KeyboardKey.c,
+    markInPoint = KeyboardKey.i,
+    markOutPoint = KeyboardKey.o,
+    scrubForward = KeyboardKey.k,
+    scrubBackward = KeyboardKey.j,
+}
+
+const StyledAudioPlayer = styled('audio')`
+    border-radius: 20px;
+`;
+
+const calculateTimeRangeSelectionStatus = (
+    inPointSeconds: number | null,
+    outPointSeconds: number | null
+): TimeRangeSelectionStatus => {
+    if (isNull(inPointSeconds)) return `noSelection`;
+
+    if (isNull(outPointSeconds)) return `inPointOnly`;
+
+    return `fullSelection`;
+};
+
+export type TimeRangeSelection = {
+    inPointSeconds: number;
+    outPointSeconds: number;
+};
+
+interface AudioAnnotatorProps {
+    audioUrl: string;
+    mimeType?: AudioMIMEType;
+    onTimeRangeSelected: (timeRangeSelected: Nullable<TimeRangeSelection>) => void;
+}
+
+export const AudioAnnotator = ({
+    audioUrl,
+    mimeType,
+    onTimeRangeSelected,
+}: AudioAnnotatorProps) => {
+    const audioRef = useRef<HTMLAudioElement>(null);
+
+    const [inPointSeconds, setInPointSeconds] = useState<Nullable<number>>(null);
+
+    const [outPointSeconds, setOutPointSeconds] = useState<Nullable<number>>(null);
+
+    const [isPlayable, setIsPlayable] = useState<boolean>(false);
+
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    const timeRangeSelectionStatus = calculateTimeRangeSelectionStatus(
+        inPointSeconds,
+        outPointSeconds
+    );
+
+    const blurAudioPlayer = () => {
```

Comment:
This is what I could figure out as a hack for the issue with the focus remaining on the `<audio>` element after scrubbing.  (This was not an issue in Firefox, only Chrome)  But now, the user can set an in-point, use the default player controls to scrub to a later point and then use a hot key to set the out-point without running into trouble.

I couldn't find a way to test this because I can't find a way to actually `click()` one of the `<audio>` controls (buttons or scrub bar) in Cypress.  I even tried clicking on coordinates within the audio player but I guess the audio element is not actionable so just won't click.

Reply:
You might want to add a comment about this in the actual code.